KeyVaultClientEncryption/keyvault_cse_migration.py

```diff
@@ -0,0 +1,130 @@
+    CustomerProvidedEncryptionKey, BlobType, BlobProperties
+from azure.keyvault.keys import KeyClient
+from azure.identity import ClientSecretCredential
+from azure.keyvault.keys.crypto import CryptographyClient, EncryptionAlgorithm
+
+
+def get_key(key_client, keyname):
+    # get key from keyvault using name of the key
+    # this key must already exist in keyvault
+    key = key_client.get_key(keyname)
+    # return keyvault key
+    return key
+
+
+def download_blob(bs_client, cont_name, blob_name):
+    # download blob and contents of blob-- return contents in bytes
+    print("Downloading blob from azure...\n")
+    # access specific container and blob using blob client that you want to download from
+    blob_client = bs_client.get_blob_client(container=cont_name, blob=blob_name)
+    # write contents from blob into a file as bytes and download
+    with open(blob_name, "wb+") as download_file:
+        download_file.write(blob_client.download_blob().readall())
+    # access contents from file and read
+    file = open(blob_name, "rb")
+    content = file.read()
+    file.close()
+
+    # return contents from file in bytes
+    return content
+
+
+def decrypt_blob(key, credential, content, blob_name):
+    # decrypt blob-- requires data in form of bytes
+    print("Decrypting blob...")
+    # use keyvault key and credential to access account client for decryption
+    crypto_client = CryptographyClient(key, credential=credential)
+    # decrypt content
+    decrypted = crypto_client.decrypt(EncryptionAlgorithm.rsa_oaep, content)
+    # convert decrypted content to plaintext to make it readable
+    message = decrypted.plaintext
+
+    # write decrypted content to a file
+    file = open(blob_name, "wb+")
+    file.write(message)
+    file.close()
+
+    # return decrypted content
+    return message
+
+
+def encryption_scope(key_uri):
+    # makes a customer managed-key encryption scope
+    os.system(
+        'cmd /c "az storage account encryption-scope create --account-name ' + keyvaultConfig.STORAGE_ACCOUNT + ' --name ' + keyvaultConfig.CUSTOMER_SCOPE_NAME + ' --key-source Microsoft.KeyVault --resource-group ' + keyvaultConfig.RESOURCE_GROUP + ' --subscription ' + keyvaultConfig.SUB_ID + ' --key-uri ' + key_uri + '"')
+    # makes a Microsoft managed-key encryption scope
+    os.system(
+        'cmd /c "az storage account encryption-scope create --account-name ' + keyvaultConfig.STORAGE_ACCOUNT + ' --name ' + keyvaultConfig.SERVER_SCOPE_NAME + ' --key-source Microsoft.Storage --resource-group ' + keyvaultConfig.RESOURCE_GROUP + ' --subscription ' + keyvaultConfig.SUB_ID + '"')
+
+
+# reupload blob without client side encryption
+# this should upload as three different files
+# one that utilizes a customer managed key
+# with encryption scope, one that uses a
+# Microsoft managed key with encryption-scope,
+# and one that uses a customer provided key
+def reupload_blob(bs_client, cont_name, blob_name, message):

```

Comment:
So this file is not what you expect the customer to use right? then move it to another folder. Else customer would be confused about which one is for them to use.